mlora/pipeline/queue.py

```diff
@@ -0,0 +1,99 @@
+from mlora.pipeline.messages import PipeMessage
+from mlora.pipeline.stream import CudaStream
+
+import torch
+import logging
+
+from queue import Queue
+from threading import Thread
+from typing import Optional
+
+
+class DeviceSwapQueue:
```

Comment:
any test on this part? how to measure the pef for this one ?

Reply:
only use the nsys to test the perf, hard to add it to ci.